pkg/logforwarding/interactive.go

```diff
@@ -0,0 +1,156 @@
+var podGroupsOptions = map[string]string{
+	"API":                "kube-apiserver, openshift-apiserver",
+	"Authentication":     "oauth-server, oauth-apiserver",
+	"Controller Manager": "kube-controller-manager, openshift-controller-manager, openshift-route-controller-manager",
+	"Scheduler":          "kube-scheduler",
+}
+
+func InteractiveLogForwardingConfig(mainHelpMsg string) ([]*ocm.LogForwarderConfig, error) {
+	con, err := interactive.GetOption(interactive.Input{
+		Question: "Enabled log forwarding",
+		Help:     mainHelpMsg,
+		Required: false,
+		Default:  false,
+		Options:  []string{cloudWatch, s3, both},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if con == skip {
+		return nil, nil
+	}
+
+	result := make([]*ocm.LogForwarderConfig, 0)
+
+	if con == cloudWatch || con == both {
+		result = append(result, &ocm.LogForwarderConfig{})
+		err = interactiveCloudWatch(result[0])
+		if err != nil {
+			return nil, err
+		}
+	}
+	if con == s3 || con == both {
+		result = append(result, &ocm.LogForwarderConfig{})
+		err = interactiveS3(result[len(result)-1])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return result, nil
+}
+
+func interactiveCloudWatch(config *ocm.LogForwarderConfig) error {
+	roleArn, err := interactive.GetString(interactive.Input{
+		Question: "CloudWatch Log forwarding role ARN",
+		Help:     "The role ARN which forwards logs to CloudWatch",
+		Required: true,
+	})
+	if err != nil {
+		return err
+	}
+	config.CloudWatchLogRoleArn = roleArn
+
+	groupName, err := interactive.GetString(interactive.Input{
+		Question: "CloudWatch log group name",
+		Help:     "The name of the group on CloudWatch which will contain the logs",
+		Required: true,
+	})
+	if err != nil {
+		return err
+	}
+	config.CloudWatchLogGroupName = groupName
+
+	podGroups, err := promptForPodGroups("CloudWatch")
+	if err != nil {
+		return err
+	}
+	config.GroupsLogVersion = podGroups
+
+	applications, err := promptForApplications("CloudWatch")
+	if err != nil {
+		return err
+	}
+	config.Applications = strings.Split(",", applications)
+
+	return nil
+}
+
+func interactiveS3(config *ocm.LogForwarderConfig) error {
+	bucketPrefix, err := interactive.GetString(interactive.Input{
+		Question: "S3 Bucket prefix",
+		Help:     "The identifiable prefix to prepend to the S3 Bucket used for log forwarding",
+		Required: true,
+	})
+	if err != nil {
+		return err
+	}
+	config.S3ConfigBucketPrefix = bucketPrefix
+
+	bucketName, err := interactive.GetString(interactive.Input{
+		Question: "S3 Bucket name",
+		Help:     "The identifiable name to append to the S3 Bucket used for log forwarding",
+		Required: true,
+	})
+	if err != nil {
+		return err
+	}
+	config.S3ConfigBucketName = bucketName
+
+	podGroups, err := promptForPodGroups("S3")
+	if err != nil {
+		return err
+	}
+	config.GroupsLogVersion = podGroups
+
+	applications, err := promptForApplications("S3")
+	if err != nil {
+		return err
+	}
+	config.Applications = strings.Split(",", applications)
```

Comment:
It should be config.Applications = strings.Split(applications,",") to split the applications